chitchat/src/lib.rs

```diff
@@ -364,6 +356,23 @@ impl Chitchat {
         };
 
         if node_state.max_version() >= max_version {
+            info!("attempted to reset node, but node is already up to date");
+            return;
+        }
+
+        if max_version < node_state.last_gc_version() {
+            // It is possible to have gc_version > max_version when we are catching up.
+            // If we reach this point, we probably went through a reset via chitchat gossip.
+            //
+            // Now we are trying to reset our state via grpc gossip, but the new state is already
+            // be out of date.
+            warn!(
```

Comment:
this is the new condition.

Before, when we gossiped with lagging nodes, we would update our state, and trigger gossip in the same round.